src/services/api/chat-guidance/dto/chat.guidance.query.result.dto.ts

```diff
@@ -0,0 +1,17 @@
+import { Field, ObjectType } from '@nestjs/graphql';
+import { IChatGuidanceResult } from './chat.guidance.result.dto';
+
+@ObjectType('ChatGuidanceQueryResult')
+export abstract class IChatGuidanceQueryResult extends IChatGuidanceResult {
```

Comment:
I do not understand the design with the two DTO's. Not sure why one inherits from the other, and the base one is used in various places where it does not make sense as far as I can see.

Reply:
I left it like that because the genai api returns the result as such - with payload { result: ... } and the query returns a superset of that. It is fit-for-purpose, at least between the gen ai API - rabbitMQ - Alkemio Server. I doubt it is needed on Alkemio API level, hence the stories raised - but merits a discussion.